src/client.rs

```diff
@@ -130,30 +130,37 @@ impl Client {
 		Ok(())
 	}
 
-	/// List all available workspaces, also filtering between those owned and those invited to.
-	pub async fn list_workspaces(&self, owned: bool, invited: bool) -> RemoteResult<Vec<String>> {
-		let mut workspaces = self
+	/// Fetch the names of all workspaces owned by the current user.
+	pub async fn fetch_owned_workspaces(&self) -> RemoteResult<Vec<String>> {
+		self.fetch_workspaces(true).await
+	}
+
+	/// Fetch the names of all workspaces the current user has joined.
+	pub async fn fetch_joined_workspaces(&self) -> RemoteResult<Vec<String>> {
+		self.fetch_workspaces(false).await
+	}
+
```

Comment:
i don't think it is necessary and it leads to confusion. editors should never just get the whole list: they need to visually differentiate between owned and invited workspaces, because owned ones can have extra buttons (or actions run on them in general).

Also `owned_workspaces` will probably always be few, while `joined_workspaces` may be a lot and require partial responses or longer waits.

Last, these should not be used except when attaching the first time and when rejoining after a network error: to have "realtime" invite notifications we will eventually introduce a Session-level stream, which allows to keep a cached list of workspaces which gets updated as needed when events are received. The cached list should be editor's authority as it will be possible to keep it perfectly synched with events (just like the buffer list)

Reply:
In sublime when dealing with multiple workspaces and invoking commands such as `attach_buffer` I also need to know to which workspace a user wants to attach to and so I present him with a list, which contains both owned and invited workspaces.
Not having the ability to visually interact and present different kinds of workspaces I am forced to ask the user and can't properly infer it.

This is nothing major though, can easily work around it.

Reply:
we figured that generally, even if you need both, you need them separate, because it would be desirable to show slightly differently (highlighting them somehow) the ones that are owned, since you have more permissions on them